src/lib/components/organisms/scanner/tools.ts

```diff
@@ -33,38 +46,34 @@ function isUrlAllowed(url: string): boolean {
 export const parseQr = (value: string): ParseQrResults => {
 	const notValidQr = 'not valid qr';
 	let parsedValue: Record<string, unknown>;
+	let type: 'credential' | 'service';
 	try {
 		parsedValue = JSON.parse(value);
 	} catch (e) {
 		return { result: 'error', message: notValidQr };
 	}
-	try {
-		credentialSchema.parse(parsedValue);
-	} catch (e) {
+	if (credentialSchema.safeParse(parsedValue).success) {
+		type = 'credential';
+		parsedValue.type = 'credential';
+	} else if (serviceSchema.safeParse(parsedValue).success) {
+		type = 'service';
+		parsedValue.type = 'service';
+	} else {
 		return { result: 'error', message: notValidQr };
 	}
-	if (!isUrlAllowed(parsedValue.url as string)) {
+
+	if (type == 'service' && !isUrlAllowed(parsedValue.url as string)) {
 		return { result: 'error', message: 'not allowed verifier url' };
 	}
-	return { result: 'ok', credential: parsedValue as Credential };
+	//todo: validate service urls
+	if (type == 'service') {
+		return { result: 'ok', data: { type, service: parsedValue as Service } };
+	} else {
+		return { result: 'ok', data: { type, credential: parsedValue as Credential } };
+	}
```

Comment:
This should come from the credential-issuer/.well-known

Reply:
can you explalin me better?